### PR TITLE
bugfix: header rewrite empty

### PR DESCRIPTION
**Pull request by zhu327**
### Description

Fixes # (issue)

### Checklist

- [ ] 填写 PR 描述及相关 issue (write PR description and related issue)
- [ ] 代码风格检查通过 (code style check passed)
- [ ] PR 中包含单元测试 (include unit test)
- [ ] 单元测试通过 (unit test passed)
- [ ] 本地开发联调环境验证通过 (local development environment verification passed)

### CODE REVIEW

**Review by wklken — APPROVED** [review]
LGTM

File: src/apisix/t/bk-stage-header-rewrite.t [hunk 1]
```diff
@@ -105,7 +105,8 @@ passed
                         "bk-stage-header-rewrite": {
                             "set": {
                                 "X-Api-Version": "v2"
-                            }
+                            },
+                            "remove": []
                         }
```
Comment (wklken) [inline src/apisix/t/bk-stage-header-rewrite.t:110]: 需要加个case, 验证 set/add 为空, remove有值的场景

Comment (zhu327) [reply src/apisix/t/bk-stage-header-rewrite.t:110]: Done

File: src/apisix/t/bk-stage-header-rewrite.t [hunk 2]
```diff
@@ -355,3 +355,42 @@ uri: /uri/plugin_proxy_rewrite
...
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- request
+GET /t
+--- response_body
+passed
```
Comment (wklken) [inline src/apisix/t/bk-stage-header-rewrite.t:396]: 上面只代表注册成功(check schema通过), 还需要测试访问正常

```
=== TEST 14: remove header set empty ok
--- request
GET /hello HTTP/1.1
--- more_headers
X-Api-Test: foo
X-Api-Engine: bar
--- response_body
uri: /uri/plugin_proxy_rewrite
host: localhost
x-api-engine: APISIX
x-real-ip: 127.0.0.1
```

Comment (zhu327) [reply src/apisix/t/bk-stage-header-rewrite.t:396]: Done

**Pull request merged**